### DISCUSSION TITLE
Renovate and Copier

### DISCUSSION THREAD

**Discussion by JanEricNitschke** [renovatebot/renovate#33191] [Request Help]
### How are you running Renovate?

None

### Please tell us more about your question or problem

I am using [Copier](https://copier.readthedocs.io/en/stable/) to create a template for a python project using poetry.

I would now like to ensure that the jinja template files of the pyproject.toml and .pre-commit-config.yaml are up to date.

Is renovate able to handle something like that? I [saw](https://docs.renovatebot.com/modules/manager/copier/) that it supports copier to some degree but i think it is just that a project produced from a copier template can ensure that it is in sync with the latest version of the templates it was created from and now to ensure that the templates always have the dependencies up to date.

**Comment by reitzig** [comment]
Definitely possible.

That should be possible by [extending the respecive manager's `fileMatch`](https://docs.renovatebot.com/modules/manager/#file-matching) to tell Renovate to care about your `.jinja` files at all. 
(FWIW, I don't see a way to do that through `packageRules`.)

If the included managers can't handle the template files -- they may not be valid YAML or whatever, after all! -- you will have to use [`customManagers`](https://docs.renovatebot.com/configuration-options/#custommanagers), which is an entire experience. Best of luck!

**Reply by JanEricNitschke** [reply]
Thanks. I will look into this if a find the time. For how i just opted to instantiate the project, run the upgrade tools on that and then move the changes back into the template. Very hacky, but works and now i got too much to do. But will definitely keep this in my mind.

**Comment by lkubb** [comment]
There's also the possibility to extract the version declarations from your `pyproject.toml`/`.pre-commit-config.yaml` into a dedicated, non-templated file and rely on Copier to render the versions:

https://github.com/salt-extensions/salt-extension-copier/blob/main/data/versions.yaml
https://github.com/salt-extensions/salt-extension-copier/blob/main/project/.pre-commit-config.yaml.j2

**Comment by ulgens** [comment]
I have a similar case with https://github.com/ulgens/python-anything/ . I want to update dependencies in Jinja-flavored dep / config files, for example https://github.com/ulgens/python-anything/blob/821b47c416191d162e3008b2e15685368bae1f1e/template/.pre-commit-config.yaml.jinja 

Creating a custom manager / regex parser is possible but feels a bit too ugly. salt-extension's approach from @lkubb 's comment above seems ok, but because it still requires some customization, wouldn't be my first choice. https://github.com/renovatebot/renovate/discussions/35466 seems a bit more interesting: Rendering the Jinja file with selected inputs, before the renovate's execution, and then somehow applying the found changes to the original Jinja file would help a lot.
